scripts/globals/job_utils/ranger.lua

```diff
@@ -0,0 +1,277 @@
+
+    if ranged and ranged:isType(xi.itemType.WEAPON) then
+        local skilltype = ranged:getSkillType()
+        if
+            skilltype == xi.skill.ARCHERY or
+            skilltype == xi.skill.MARKSMANSHIP or
+            skilltype == xi.skill.THROWING
+        then
+            if
+                ammo and
+                (
+                    ammo:isType(xi.itemType.WEAPON) or
+                    skilltype == xi.skill.THROWING
+                )
+            then
+                ability:setRecast(math.max(0, ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST) * 60))
+                return 0, 0
+            end
+        end
+    end
+
+    return xi.msg.basic.NO_RANGED_WEAPON, 0
+end
+
+xi.job_utils.ranger.checkVelocityShot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkSharpshot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkScavenge = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkCamouflage = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkBarrage = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkShadowbind = function(player, target, ability)
+    if
+        (player:getWeaponSkillType(xi.slot.RANGED) == xi.skill.MARKSMANSHIP and
+        player:getWeaponSkillType(xi.slot.AMMO) == xi.skill.MARKSMANSHIP) or
+        (player:getWeaponSkillType(xi.slot.RANGED) == xi.skill.ARCHERY and
+        player:getWeaponSkillType(xi.slot.AMMO) == xi.skill.ARCHERY)
+    then
+        return 0, 0
+    end
+
+    return 216, 0 -- You do not have an appropriate ranged weapon equipped.
+end
+
+xi.job_utils.ranger.checkUnlimitedShot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkFlashyShot = function(player, target, ability)
+    return 0, 0 -- Not implemented yet
+end
+
+xi.job_utils.ranger.checkStealthShot = function(player, target, ability)
+    return 0, 0 -- Not implemented yet
+end
+
+xi.job_utils.ranger.checkDoubleShot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkBountyShot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkDecoyShot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkHoverShot = function(player, target, ability)
+    return 0, 0  -- Not implemented yet
+end
+
+xi.job_utils.ranger.checkOverkill = function(player, target, ability)
+    ability:setRecast(math.max(0, ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST) * 60))
+
+    return 0, 0
+end
+
+-----------------------------------
+-- Ability Use Functions
+-----------------------------------
+
+xi.job_utils.ranger.useEagleEyeShot = function(player, target, ability)
+ if player:getWeaponSkillType(xi.slot.RANGED) == xi.skill.MARKSMANSHIP then
+        action:setAnimation(target:getID(), action:getAnimation(target:getID()) + 1)
```

Comment:
Did you test this, `action` doesn't exist in this scope?

Reply:
I had a test this morning because I was 100% I did test each skill against an enemy. **HOWEVER** you are 100% this is mistake on my part, the skill will trigger and fire the arrow, however no damage nor any "action" is taken against the monster itself but the skill will go on cooldown. This was my inexperience and not being careful enough. Will fix